src/dlgRoomExits.cpp

```diff
@@ -328,8 +328,8 @@ void dlgRoomExits::save()
             pR->setExitWeight(exitKey, 0);
         }
     } else { // No valid exit on the dialogue
-        if (originalExits.value(dirCode)->destination > 0) {
-            pR->setExit(-1, dirCode); // Destination has been deleted So ensure the value for no exit is stored
+        if (originalExits.contains(dirCode) && originalExits.value(dirCode)->destination > 0) {
```

Comment:
Why check twice? Do you think it's faster?

Reply:
Using `TMap<T1, T2>::value(const T1 key)` will return a default constructed `nullptr` value should the key not be present when T2 is a pointer type. So the first test is needed to confirm that there will be a valid (non-nullptr) pointer for the second test to dereference. The absence of that test was what Coverity was unhappy about...